CONTRIBUTING.md

```diff
@@ -1,60 +1,129 @@
-# Typical development workflow
+# Contributing
 
-Fork the repo, work on an issue
+This guide explains how to set up your environment, make changes, and submit a PR.
 
-## Updating the generated Kibana client.
+## Development Setup
 
-If your work involves the Kibana API, the endpoints may or may not be included in the generated client.
-Check [generated/kbapi](./generated/kbapi/) for more details.
+* Fork and clone the repo.
+* Setup your preferred IDE (IntelliJ, VSCode, etc.)
 
-## Acceptance tests
+Requirements:
+* [Terraform](https://www.terraform.io/downloads.html) >= 1.0.0
+* [Go](https://golang.org/doc/install) >= 1.25
+* Docker (for acceptance tests)
 
-```bash
-make docker-testacc
-```
-
-Run a single test with terraform debug enabled:
-```bash
-env TF_LOG=DEBUG make docker-testacc TESTARGS='-run ^TestAccResourceDataStreamLifecycle$$'
-```
-
-A way to forward debug logs to a file:
-```bash
-env TF_ACC_LOG_PATH=/tmp/tf.log TF_ACC_LOG=DEBUG TF_LOG=DEBUG make docker-testacc
-```
+## Development Workflow
 
+* Create a new branch for your changes.
+* Make your changes. See [Useful Commands](#useful-commands) and [Debugging](#running--debugging-the-provider).
+* Validate your changes
+  * Run unit and acceptance tests (See [Running Acceptance Tests](#running-acceptance-tests)).
+  * Run `make lint` to check linting and formatting. For this check to succeed, all changes must have been committed.
+  * All checks also run automatically on every PR.
+* Submit your PR for review.
+* Add a changelog entry in `CHANGELOG.md` under the `Unreleased` section. This will be included in the release notes of the next release. The changelog entry references the PR, so it has to be added after the PR has been opened.
 
-## Update documentation
+When creating new resources:
+* Use the [Plugin Framework](https://developer.hashicorp.com/terraform/plugin/framework/getting-started/code-walkthrough) for new resources.
+  * Use an existing resource (e.g. `internal/elasticsearch/security/system_user`) as a template.
+  * Some resources use the deprecated Terraform SDK, so only resources using the new Terraform Framework should be used as reference.
+* Use the generated API clients to interact with the Kibana APIs. (See [Working with Generated API Clients](#working-with-generated-api-clients)
+* Add a documentation template and examples for the resource. See [Updating Documentation](#updating-documentation) for more details.
+* Write unit and acceptance tests.
 
-Update documentation templates in `./templates` directory and re-generate docs via:
-```bash
-make docs-generate
-```
+### Useful Commands
 
-## Update `./CHANGELOG.md`
+* `make build`: Build the provider.
+* `make lint`: Lints and formats the code.
+* `make test`: Run unit tests.
+* `make docs-generate`: Generate documentation.
```

Comment:
It makes sense to run acceptance tests locally too. Unlike EC provider, the stack provider can run acceptance tests against locally running ES and Kibana, e.g. please refer to `make docker-testacc` (it starts the apps in docker)  or `make testacc`. The later one requires already running ES and Kibana (they can run in docker) and env vars pointing to them.